### PR TITLE
Add survey url card

### LINKED ISSUE #1011: Survey URL card
## Description
A user should be able to find the link to a survey easily.

## Relevant Job Stories
* When my survey is ready for responses, I want to share a link to where it can be filled out online, so that users can easily submit their responses

## Prerequisites
None

## Requirements
- [ ] There should be a smallish card on the overview page
- [ ] The card always contains a text field with the public URL for the survey
- [ ] A copy button copies the link to the clipboard
- [ ] A clickable link opens the public survey page in a new tab, at the url ``/o/${orgId}/surveys/${surveyId}`` (which will automatically be redirected by NEXT to the correct URL)
- [ ] The card has two states
- [ ] When the survey is active, display it's "active" state
  - [ ] Headline is "Open for submissions"
  - [ ] Caption is "Now accepting submissions at this link"
  - [ ] There is a green status indicator
  - [ ] The link reads "Visit survey in activist portal"
- [ ] When the survey is not active, display it's "preview" state
  - [ ] Headline is "Preview survey"
  - [ ] Caption is "Will accept submissions at this link"
  - [ ] There is a gray status indicator
  - [ ] The link reads "Preview survey in activist portal"

## Possible implementations
See `ZUICopyToClipboard` and `ZUIInlineCopyToClipboard` for code to copy text to clipboard, and consider refactoring to create something more reusable.

### Active state in situ
This task only concerns the smallish card in the top right called "Open for submissions" in the mockup below.

image

### Draft/preview state
image

## Open questions
None

### PR DESCRIPTION
**Pull request by danbjork**
## Description
This PR adds a component `SurveyURLCard.tsx`, its translations and a new component `TextfieldToClipboard.tsx`

## Screenshots
Screenshot 2023-03-01 at 13 58 14
Screenshot 2023-03-01 at 13 58 06

## Changes
* Adds translations
* Adds global public shareable URL to config
* Adds SurveyURLCard.tsx

## Notes to reviewer
- Do we want a test for this component? Let me know and I will write one!

## Related issues
Resolves #1011

### CODE REVIEW

File: src/zui/ZUITextfieldToClipboard.tsx
Comment (danbjork) [inline src/zui/ZUITextfieldToClipboard.tsx:22]: Perhaps not that necessary... I could remove its since the setTimeout resets `copied`s value

Comment (richardolsson) [reply src/zui/ZUITextfieldToClipboard.tsx:22]: Does this even do anything? When the dependency array is empty, the effect will run when the component is mounted. When a component is mounted, it's state is already at it's default state, i.e. `false` in this case.

I think you did this to handle situations where the user navigates away and comes back, to make sure that `copied` is false when they return. But in that case, the component will have unmounted (when navigating away) and when it remounts, the default state is set on line 10.

File: src/features/surveys/components/SurveyURLCard.tsx
Comment (richardolsson) [inline src/features/surveys/components/SurveyURLCard.tsx:14]: I believe the card is missing the little green/gray dot in the top right?

**Review by richardolsson — CHANGES_REQUESTED** [review]
Awesome work on this! I think the code looks great, except for the `useEffect()` that you pinpointed yourself. Also, I believe it's missing a small design element – a red/gray dot in the top right corner.

Are you able to fix this in the next few days @danbjork? No worries if not!

File: src/zui/ZUITextfieldToClipboard.tsx
Comment (danbjork) [reply src/zui/ZUITextfieldToClipboard.tsx]: You are right! Thought about it for a bit and then tested just to be sure

File: src/features/surveys/components/SurveyURLCard.tsx
Comment (danbjork) [reply src/features/surveys/components/SurveyURLCard.tsx:14]: Yes its missing! I added an indicator now. Here are different variations of `isOpen`

Screenshot 2023-03-01 at 11 36 51
Screenshot 2023-03-01 at 11 36 42

File: src/features/surveys/components/SurveyURLCard.tsx
Comment (danbjork) [inline src/features/surveys/components/SurveyURLCard.tsx:25]:
```
  backgroundColor: isOpen ? Color.success : Color.danger
```
Do we have any default colors that could be used here? I'm thinking some generic value, like the example I gave above `Colors.success` 🤔

Comment (danbjork) [reply src/features/surveys/components/SurveyURLCard.tsx]: Great! I found some colors that look similar to the Figma mockups
```
              ? theme.palette.success.main
              : theme.palette.grey['500'],
```
[0b23a8b](https://github.com/zetkin/app.zetkin.org/pull/1054/commits/0b23a8bacad8584575ada3674519d43811f93ef9)

**Review by richardolsson — DISMISSED** [review]
Nice work! I feel like this is ready to merge now. 💯 

Tomorrow we are likely to merge the internationalization refactor (#1048). That will require the strings in this PR to change. We can do it in one of two ways, and I would like to hear which one you prefer @danbjork.

A. We merge your PR now, and I migrate your code to use the new strings when I merge #1048 tomorrow.
B. We hold off on merging this for now, and once we've merged #1048 you can make the changes to your code yourself.

Whichever is fine by me! I just wanted to give you the opportunity to try out the new message format on your own code, if you want.

**Review by richardolsson — APPROVED** [review]
Merge conflicts successfully resolved. Time to merge. Great work on this @danbjork!

**Pull request merged**
